apps/windows_explorer/windows_explorer.py

```diff
@@ -1,7 +1,51 @@
+"""
+Windows Explorer Integration for Talon Voice Control
+
+This module provides voice control integration for Windows File Explorer by:
+
+1. LANGUAGE DETECTION & LOCALIZATION:
+   - Detects system UI language (English, German, etc.)
+   - Maps localized folder names to actual filesystem paths
+   - Handles different Explorer window title formats per language
+
+2. WINDOW TITLE PROCESSING:
+   - Strips Explorer-specific suffixes (e.g., "- File Explorer", multi-tab indicators)
+   - Uses regex patterns to clean window titles for path extraction
+   - Supports both single and multi-tab Explorer windows
+
+3. ONEDRIVE INTEGRATION:
+   - Detects OneDrive installation and folder structure
+   - Maps standard folders (Desktop, Documents, Pictures) to OneDrive paths when applicable
+   - Handles hybrid OneDrive/local folder scenarios
+
+4. PATH MAPPING & EXCLUSIONS:
+   - Maps localized display names to actual filesystem paths
+   - Excludes system dialogs and non-navigable windows
+   - Handles user display name mapping for personalized folders
+
+5. VOICE COMMAND ACTIONS:
+   - Navigation (open parent, go to directory)
+   - File operations (new folder, open file, properties)
+   - Address bar control (focus, copy, navigate)
+   - Terminal integration (open command prompt in current location)
+
+This has been tested with:
+   - Windows 11 24H2, Version 10.0.26100 Build 26100, English and German system language
+"""
+
+import logging
 import os
+import re
 
 from talon import Context, Module, actions, app, ui
 
+# Configuration
+DEBUG_LOGGING = False
+
+# Configure logging
+if DEBUG_LOGGING:
+    logging.basicConfig(level=logging.DEBUG, format="%(levelname)s: %(message)s")
```

Comment:
If you want to declare a logger that is private to your module you can use something like https://github.com/talonhub/community/blob/14e89d69cfb8b59a0e7f403838c352249fa0c9af/core/command_client/rpc_client/rpc_client.py#L13. Then generally you can just set the log level based on the Talon setting as above and the overhead from constructing log messages will be negligible when compared with the verbosity repeated checks add to your code.